packages/cojson/src/sync.ts

```diff
@@ -285,27 +291,85 @@ export class SyncManager {
         logger.error("Error sending new content piece, retrying", { err: e });
         peer.optimisticKnownStates.dispatch({
           type: "SET",
-          id,
-          value: optimisticKnownStateBefore ?? emptyKnownState(id),
+          id: coValue.id,
+          value: optimisticKnownStateBefore ?? emptyKnownState(coValue.id),
         });
-        return this.sendNewContentIncludingDependencies(id, peer);
+        return this.sendNewContent(coValue, peer);
       });
 
       peer.optimisticKnownStates.dispatch({
         type: "COMBINE_WITH",
-        id,
+        id: coValue.id,
         value: coValue.knownState(),
       });
     }
   }
 
-  addPeer(peer: Peer) {
+  async startPeerReconciliation(peer: PeerState) {
```

Comment:
Before we were sending the content without waiting for the `known` state from the server.

Done some changes so we just send a `load` request for each available coValue to the server.

Added some logic to make the server request the content if they notice that they miss some content from the client.